### DISCUSSION TITLE
Help Needed

### DISCUSSION THREAD

**Discussion by DevBadger10** [CosmosOS/Cosmos#2407] [Q&A]
I have been writing my own OS off of Poncho's OSDev tutorial, but it's hard to get inspiration for working on it. I think an easier way of making the same thing would help with this.

I have tried to install off the links on the website, but one of the Microsoft ones isn't found. Would I be able to have working links for all the things? And I am also a complete noob when it comes to C / C# / C+ / C++ / whatever other Cs are out there, so could I maybe have some tutorial links that you guys think are good?

Thanks for your time!

**Comment by xyve7** [comment]
Just to clarify, do you want help with Cosmos, or general OSDEV? Or, do you want pointers on getting started with the C language family?

**Reply by DevBadger10** [reply]
I would like links to make COSMOS work, and tutorials on C# so I can use that knowledge to better make use of expanding the COSMOS tutorials into my own OS.

**Reply by AsertCreator** [reply]
> I would like links to make COSMOS work, and tutorials on C# so I can use that knowledge to better make use of expanding the COSMOS tutorials into my own OS.

you need to learn c# and basic concepts of dotnet, just google "learn c#" and "what is dotnet". then (maybe one of c# tutorials already said to do that) install microsoft visual studio 2019 or 2022. you need a windows installed, mac versions of visual studio or visual studio code doesnt work. after that follow userkit installation instructions on cosmos site https://www.gocosmos.org/ (i cant put html link). keep in mind: not all of c# and dotnet features are available in cosmos, and will crash either at compilation or at runtime. also try to make some c# programs that run on windows first, to have some real look of language.
its not that hard, c# is really easy language and osdev is not so creepy. all you need is being creative making your os.

edit: you are trying to make os, when you didnt write any c# program, arent you? thats not good.

**Reply by DevBadger10** [reply]
> > I would like links to make COSMOS work, and tutorials on C# so I can use that knowledge to better make use of expanding the COSMOS tutorials into my own OS.
> 
> you need to learn c# and basic concepts of dotnet, just google "learn c#" and "what is dotnet". then (maybe one of c# tutorials already said to do that) install microsoft visual studio 2019 or 2022. you need a windows installed, mac versions of visual studio or visual studio code doesnt work. after that follow userkit installation instructions on cosmos site https://www.gocosmos.org/ (i cant put html link). keep in mind: not all of c# and dotnet features are available in cosmos, and will crash either at compilation or at runtime. also try to make some c# programs that run on windows first, to have some real look of language. its not that hard, c# is really easy language and osdev is not so creepy. all you need is being creative making your os.
> 
> edit: you are trying to make os, when you didnt write any c# program, arent you? thats not good.

I know it is not so creepy, as I have some experience. When you said "you are trying to make os, when you didnt write any c# program, arent you? thats not good.", were you meaning "Make one to gain experience with the language" or "Make a prototype"? If you don't respond, I'll guess it's both of them.

Edit: In this I personally think I sound kinda condescending, like "Oh, you didn't know this"? But that wasn't the intention of the tone at all! I don't know, maybe it's just my brain, but I want you to know that wasn't the intention of the tone.

**Reply by AsertCreator** [reply]
I meant that as far as I know, you didnt write c# userland programs. userland programs are running under some os, like windows. osdev is very hard, you will likely to be confused with terms or some "cpu hacks". this is what meant under "you are trying to make os, when you didnt write any c# program"

p.s. i didnt notice your tone intention

edit: forgot to mention that cosmos is not THAT popular platform to have tutorials on google search
